### DISCUSSION TITLE
MJPEG Server - Question

### DISCUSSION THREAD

**Discussion by Wolzo** [raspberrypi/picamera2#227] [Q&A]
Hi everyone, I'm not sure if this is the right place to ask such questions but I've got a problem
I'm currently updating an old project, it was done with the old libraries (raspistill and picamera). I've upgraded both the system and the hardware and they work fine, the only problem is with my python server. With the old libraries I had a nice image, but with the new ones the video stream seems zoomed. 
This "zoom" problem is present even with the libcamera commands via terminal, for example: 
if I take an image with libcamera-still everything is fine, instead recording a video with libcamera-vid produces a zoomed result

Has anyone had this problem?

**Comment by cdg123** [comment]
Hi @DrMicio.

Yes when I started looking at this a couple of days ago it confused me for a bit.

What_ I found was that the libcamera-still app defaults to capturing an image at 1640x1232 on my v2 picamera whereas the libcamera-vid defaults to 640x480 so appears zoomed as you say.

If you specify the width and height options then this sorted it for me.

Try `libcamera-vid --width 1640 --height 922` 

There's lots to learn I am finding but looks like a really useful way to get better access to everything.

I also found invaluable the --help and -v verbose options great to find where I was going wrong.

Hopefully that helps you if you haven't managed to figure it out.

**Reply by Wolzo** [reply]
Hi @cdg123,

thanks for the reply.
Adding those parameters seems to resolve the issue. But I'm still clueless on how to replicate those parameters with python. I've already tried to set the size on the main video configuration while initializing the camera object but it doesn't work

`picam2.configure(picam2.create_video_configuration(main={"size": (1640, 922)}))`

**Reply by Wolzo** [reply]
I solved this issue by changing the raw format

My code now is something like that

```
tuning = Picamera2.load_tuning_file("imx219.json")
picam2 = Picamera2(tuning=tuning)

main = {"format": "XBGR8888", "size": (480, 360)}
picam2.make_initial_stream_config(main, main)
picam2.align_stream(main)

lores = {"format": "YUV420", "size": (480, 360)}
picam2.make_initial_stream_config(lores, lores)

raw = {"format": picam2.sensor_format, "size": (1640, 1232)}
picam2.make_initial_stream_config(raw, raw)

controls = {"NoiseReductionMode": libcamera.controls.draft.NoiseReductionModeEnum.Minimal,"FrameDurationLimits": picam2.camera_controls["FrameDurationLimits"][0:1]}
config = {"use_case": "still","transform": libcamera.Transform(),"colour_space": libcamera.ColorSpace.Jpeg(),"buffer_count": 6,"main": main,"lores": lores,"raw":raw,"controls":controls}
picam2.add_display_and_encode(config, "main", "main")
picam2.configure(config)
```

Thanks for the help

**Reply by sandyol55** [reply]
With a V2 camera the request for a 1640 x 922 image size will result in sensor mode of 1920 x 1080 being used,  with its corresponding reduced FOV.

Requesting a 1640 x 1232 size will return a full 4x3 format (binned?) image size.

If the requirement is for a 16x9 type format then 
 `picam2.configure(picam2.create_video_configuration(main={"size": (1640, 922)}, raw = {"size" : (1640,1232)}))`
should work.


Ahh missed your update!!

**Reply by davidplowman** [reply]
Hi, sorry to have missed this thread. I'm not really expecting people to do things as complicated as calling `make_initial_stream_config` and then assembling the entire configuration themselves (indeed some of those functions are likely to marked in future as "internal"). Instead, calling `create_video_configuration` (or the other `create_xxx_configuration` functions) is better (as sandyol55 did). You do also have the option to edit the configuration object once it's been returned, before calling `configure`.

The v2 camera is a bit unlucky for 1080p recording - you can either have a cropped (not full field of view) image, or you can have full field of view but upscaled - you would need the HQ cam for full field of view _and_ not upscaled. The next _Picamera2_ release will have functionality for listing the available camera modes, and what field of view (and framerates) you will get.
